bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/tools/WaitForTopicValue.java

```diff
@@ -44,13 +46,19 @@
     public WaitForTopicValue(MqttBrokerConnection connection, String topic)
             throws InterruptedException, ExecutionException {
         final MqttMessageSubscriber mqttMessageSubscriber = (t, payload) -> {
-            future.complete(new String(payload));
+            try {
+                future.complete(new String(payload, "UTF-8"));
+            } catch (UnsupportedEncodingException e1) {
+                future.complete(new String(payload));
+            }
         };
-        future = future.whenComplete((r, e) -> {
+        future.whenComplete((r, e) -> {
             connection.unsubscribe(topic, mqttMessageSubscriber);
         });
 
-        subscripeFuture = connection.subscribe(topic, mqttMessageSubscriber);
+        subscribeFuture = connection.subscribe(topic, mqttMessageSubscriber);
+
+        composeFuture = subscribeFuture.thenCompose(b -> future);
```

Comment:
You probably don't need the `subscribeFuture` field anymore.
```suggestion
        composeFuture = connection.subscribe(topic, mqttMessageSubscriber).thenCompose(b -> future);
```

Reply:
:+1: